ee/server/calendar/google_calendar.go

```diff
@@ -350,6 +348,14 @@ func (c *GoogleCalendar) GetAndUpdateEvent(event *fleet.CalendarEvent, genBodyFn
 		}
 	}
 
+	// If event was deleted/cancelled, we need to stop watching it
+	if !channelStopped {
+		err = c.config.API.Stop(details.ChannelID, details.ResourceID)
+		if err != nil {
+			level.Warn(c.config.Logger).Log("msg", "stopping Google calendar event watch", "err", err)
+		}
+	}
```

Comment:
Could you explain the change? It seems there are two paths where the watching of the event is not stopped (all day events and one that says `// Delete this event to prevent confusion. This operation should be rare.
`). Were any of these two scenarios causing issues?

Reply:
The bug was when we deleted the event from Google calendar, we did not stop watching the notification channel. When event is deleted, it comes back from Google calendar with status = "cancelled". With this change, we will now be stopping the notification channel for that case.